Gives emitters wires

Time-Green opened the pull request:
# Adds 5 wires to an emitter:

- **Activate**: Activates the emitter. Cut it to prevent activation

- **Deactivate**: Disables the emitter. Cut to prevent deactivation

- **Fire**: Turns the emitter on for 5 seconds, giving 0-3 shots. Cut to disable it's firing mechanism

-_And 2 useless wires_

:cl:
rscadd: Gives emitters wires to: Activate, disable and fire. 
/:cl:

Super3222 commented:
Interesting. :+1: MERGE

ShadowDeath6 commented:
Cool

GrayRachnid commented:
butwhy?

ShadowDeath6 commented:
>why

hacking + consistency between machinery

Time-Green commented:
>Why

So I can set a trap in engi maint to kill all assistants

GrayRachnid commented:
>hacking + consistency between machinery
If this is about consistency then you've got a huge list of machinery to work on.
Let's forget the rest and focus on engineering based ones only. SMES, PACMANs, Turbine, the tesla thingies, and the other tesla thingies.

>So I can set a trap in engi maint to kill all assistants
What stops you from that now tho
inb4 you don't know that you can already lock emitters so nobody can turn off/on including the AI.

ShadowDeath6 commented:
Power units don't need wires tho, they're all-access

Time-Green commented:
Gray, it's about utility and fun. Mainly using it in combination with stuff like signallers to create cool combos.
Please give an actual reason instead if repeating "why"

matskuman5 commented:
![chrome_2017-05-09_23-32-43](https://cloud.githubusercontent.com/assets/11191939/25871352/dd8a2fc2-350f-11e7-9667-66d2b21ad4c5.png)

ShadowDeath6 commented on code/modules/power/singularity/emitter.dm:
```diff
@@ -168,6 +180,10 @@
 
+
+		if(!can_fire)
+			visible_message("<span class='danger'>The [src.name] hums!</span>","<span class='userdanger'>You feel a humming.</span>")
```
you don't feel humming, can you say hear humming / feel a vibration or something

Time-Green replied:
Sure

Time-Green commented:
I forgot to sync my commits, I'll do that tomorrow

Time-Green commented:
Synced my commits, tomorrow apparently became 9 days

Cruix commented:
In council time tomorrow is like 3 months.

Cruix commented:
BTW I'll merge this tomorrow

Cruix commented:
You cannot turn off emmitters manually, and when you screwdriver the panel closed it runtimes trying to qdel the qdel proc

Time-Green commented:
Really? I thought it did...
For futute reference, how do I know something is runtiming? I know what causes it, but not when it does

Cruix commented:
Look at the dream daemon window (the one with the green circle) and it will print a message like this:

runtime error: 
[09:30:51]bad del
[09:30:51]proc name: qdel (/proc/qdel)
[09:30:51]  source file: garbage.dm,164
[09:30:51]  usr: Adamantine Andesite (/mob/living/carbon/human)
[09:30:51]  src: null
[09:30:51]  usr.loc: the floor (117,158,5) (/turf/open/floor/plasteel/shuttle/purple)
[09:30:51]  call stack:
[09:30:51]qdel(null, 0)
[09:30:51]Emitter (/obj/machinery/power/emitter): attackby(the screwdriver (/obj/item/weapon/screwdriver), Adamantine Andesite (/mob/living/carbon/human), "icon-x=15;icon-y=27;left=1;scr...")
[09:30:51]Adamantine Andesite (/mob/living/carbon/human): ClickOn(Emitter (/obj/machinery/power/emitter), "icon-x=15;icon-y=27;left=1;scr...")
[09:30:51]Emitter (/obj/machinery/power/emitter): Click(the plating (118,159,5) (/turf/open/floor/plating), "mapwindow.map", "icon-x=15;icon-y=27;left=1;scr...")

It might not work if you run your server directly from dream maker. You can open a dream daemon window to run your server by opening the byond client and pressing crtl+D. Remember to set it to trusted and private before you hit run.

Time-Green commented:
Alrighty, fixed the mess.
Apparently it was going crazy because it had another screwdriver proc: the decon path.
Anyway, runtime and being unable to turn it off is fixed. I also removed the sprites and used the decon ones instead, they look better.
